### PR TITLE
Update using packagekit

### PR DESCRIPTION
**Pull request by kravciak**
Depend on:
https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/71
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/139

### CODE REVIEW

File: tests/x11/updates_xfce.pm [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use base "x11test";
+use testapi;
+
+sub run() {
+    x11_start_program("apper");
+
```
Comment (coolo) [inline tests/x11/updates_xfce.pm]: why apper on xfce?

Comment (kravciak) [inline tests/x11/updates_xfce.pm]: It was just first suggestion from rbrown
I will use updates from notification icon

File: products/opensuse/main.pm [hunk 2]
```diff
@@ -498,7 +498,9 @@ sub load_consoletests() {
         {
             loadtest "console/glibc_i686.pm";
         }
-        loadtest "console/zypper_up.pm";
+        if ( get_var("DESKTOP") !~ /gnome|kde|xfce|lxde/ ) {
```
Comment (okurz) [inline products/opensuse/main.pm]: to me and maybe to many future readers it is not obvious why you do this so I suggest to comment your decisions here

File: products/sle/main.pm [hunk 3]
```diff
@@ -856,7 +859,6 @@ sub load_x11tests() {
             loadtest "x11/xfce_lightdm_logout_login.pm";
         }
     }
-
```
Comment (okurz) [inline products/sle/main.pm]: that's an unrelated change

File: products/sle/main.pm [hunk 4]
```diff
@@ -1150,7 +1152,7 @@ else {
     }
     unless (load_applicationstests() || load_slenkins_tests()) {
         load_rescuecd_tests();
-        load_consoletests();
+#        load_consoletests();
```
Comment (okurz) [inline products/sle/main.pm]: I guess that is one of the reasons for the "WIP" ;-)

File: tests/x11/updates_gnome.pm [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+use testapi;
+use utils;
+
+sub start_updater() {
+	if (check_var("DISTRI", "opensuse")) {
+		script_run("pkill pk-update-icon");
+		script_run("pk-update-icon -d 3");
+		assert_and_click("updates_pk-update-icon");
+	} else {
+#		script_run("gsettings set org.gnome.settings-daemon.plugins.updates frequency-get-updates 120");
```
Comment (okurz) [inline tests/x11/updates_gnome.pm]: todo

File: tests/x11/updates_gnome.pm [hunk 6]
```diff
@@ -0,0 +1,88 @@
+# SUSE's openQA tests
+#
```
Comment (okurz) [inline tests/x11/updates_gnome.pm]: In the complete file: You are using tabs instead of 4 space indention. Fix your editor!

File: tests/x11/updates_gnome.pm [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+		} elsif (match_has_tag("updates_done-logout")) {
+			# cancel
+		} elsif (match_has_tag("updates_done-restart")) {
+			# cancel
+		}
+	}
+}
+
+1;
+# vim: set sw=4 et:
```
Comment (okurz) [inline tests/x11/updates_gnome.pm]: at least this line should be last in file

File: tests/x11/updates_kde.pm [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use base "x11test";
+use testapi;
+
+sub run() {
+    x11_start_program("apper");
+    
+
```
Comment (okurz) [inline tests/x11/updates_kde.pm]: delete blank lines in between

**Comment by okurz** [comment]
when you put up a WIP PR for review feel free to describe what is still WIP and what is not to help the reviewers

File: tests/x11/updates_xfce.pm (hunk 1 above)
Comment (sysrich) [inline tests/x11/updates_xfce.pm]: ?? I didn't mean to suggest that.. xfce should use pk-update-icon I believe.. appear is the KDE pk thingie.

Comment (kravciak) [inline tests/x11/updates_xfce.pm]: oh, sorry rbrown, I am talking about kde instead of xfce  :) - I will clean up this PR today..

File: tests/x11/updates_gnome.pm (hunk 6 above)
Comment (kravciak) [inline tests/x11/updates_gnome.pm]: fixed

**Comment by kravciak** [comment]
All fixed (I hope), please check description for needles

File: tests/x11/updates_gnome.pm [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            }
+            elsif (match_has_tag("updates_installed-logout")) {
+                assert_and_click("updates_click-cancel");
+            }
+            elsif (match_has_tag("updates_installed-restart")) {
+                assert_and_click("updates_click-cancel");
+
+                # Workaround - packagekit stays open, fixed after update
+                if (check_screen("updates_available", 5)) {
+                    record_soft_failure;
```
Comment (okurz) [inline tests/x11/updates_gnome.pm]: add a reference to a bug when you record a soft failure

File: tests/x11/updates_gnome.pm [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        assert_screen \@updates_tags, 100;
+        if (match_has_tag("updates_none")) {
+            send_key "ret";
+            last;
+        }
+        elsif (match_has_tag("updates_available")) {
+            assert_and_click("updates_click-install");
+
+            # Authenticate on SLES
+            if (check_var("DISTRI", "sle") && get_var('FLAVOR', '') =~ /^Server/) {
```
Comment (okurz) [inline tests/x11/updates_gnome.pm]: hm, don't know how others like this but distri/flavor specific checks within the test itself are frowned upon (at least by me ;-) ). Maybe just replace the assert_screen by a check_screen but then you don't ensure it really asks for authentication on SLES … dunno

File: tests/x11/updates_gnome.pm [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+                assert_screen("updates_authenticate", 10);
```
Comment (okurz) [inline tests/x11/updates_gnome.pm]: better than specifying a shorter then default timeout with good reasoning I suggest to just use the default.

**Comment by okurz** [comment]
see inline comments and remove WIP from title

File: tests/x11/updates_gnome.pm (hunk 9 above)
Comment (kravciak) [inline tests/x11/updates_gnome.pm]: I did not create a bug, talked to Martin Pluskal and response was:
There is no point creating a bug for leap 42.1 when issue is fixed by updating

Comment (okurz) [inline tests/x11/updates_gnome.pm]: then what should we do if we see the soft fail and don't even know why it fails?

Comment (kravciak) [inline tests/x11/updates_gnome.pm]: soft fail removed

**Comment by kravciak** [comment]
Please merge also needles:
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/139

**Comment by okurz** [comment]
done
